keyvault/client/src/main/java/com/azure/keyvault/SecretAsyncClient.java

```diff
@@ -0,0 +1,434 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.keyvault;
+
+import com.azure.common.ServiceClient;
+import com.azure.common.http.HttpPipeline;
+import com.azure.common.http.rest.PagedResponse;
+import com.azure.common.http.rest.Response;
+import com.azure.common.http.rest.SimpleResponse;
+import com.azure.common.http.rest.VoidResponse;
+import com.azure.common.implementation.RestProxy;
+import com.azure.keyvault.implementation.SecretAttributesPage;
+import com.azure.keyvault.models.DeletedSecret;
+import com.azure.keyvault.models.Secret;
+import com.azure.keyvault.models.SecretAttributes;
+import org.apache.commons.lang3.Validate;
+import org.reactivestreams.Publisher;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Objects;
+import java.util.function.Function;
+
+public final class SecretAsyncClient extends ServiceClient {
```

Comment:
JavaDoc? Refer to @alzimmermsft work on the configuration service pull request [here](https://github.com/Azure/azure-sdk-for-java/pull/3377)

Reply:
Updated Javadocs.
Added some code samples. 
Didn't add all. Wanted to get some feedback on the ones added first.